splice_machine/src/main/java/com/splicemachine/derby/procedures/SpliceAdmin.java

```diff
@@ -1712,30 +1713,14 @@ public static void SYSCS_GET_SESSION_INFO(final ResultSet[] resultSet) throws SQ
         resultSet[0] = new EmbedResultSet40(conn, resultsToWrap, false, null, true);
     }
 
-    private static final GenericColumnDescriptor[] runningOpsDescriptors = {
-            new GenericColumnDescriptor("UUID", DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, 40)),
-            new GenericColumnDescriptor("USER", DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, 40)),
-            new GenericColumnDescriptor("HOSTNAME", DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR, 120)),
-            new GenericColumnDescriptor("SESSION", DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.INTEGER)),
-            new GenericColumnDescriptor("SQL", DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR)),
-            new GenericColumnDescriptor("SUBMITTED", DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR,40)),
-            new GenericColumnDescriptor("ELAPSED", DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR,40)),
-            new GenericColumnDescriptor("ENGINE", DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR,40)),
-            new GenericColumnDescriptor("JOBTYPE", DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR,40)),
-    };
-
-    public static void SYSCS_GET_RUNNING_OPERATIONS(final ResultSet[] resultSet) throws SQLException {
-        List<ExecRow> rows = getRunningOperations();
+    public static void SYSCS_GET_RUNNING_OPERATIONS(final ResultSet[] resultSet) throws SQLException{
+        String sql = "call SYSCS_UTIL.SYSCS_GET_RUNNING_OPERATIONS_LOCAL()";
+        resultSet[0] = getRunningOperations(sql, false, false ).getResultSet();
+    }
 
-        EmbedConnection conn = (EmbedConnection)getDefaultConn();
-        Activation lastActivation = conn.getLanguageConnection().getLastActivation();
-        IteratorNoPutResultSet resultsToWrap = new IteratorNoPutResultSet(rows, runningOpsDescriptors, lastActivation);
-        try {
-            resultsToWrap.openCore();
-        } catch (StandardException se) {
-            throw PublicAPI.wrapStandardException(se);
-        }
-        resultSet[0] = new EmbedResultSet40(conn, resultsToWrap, false, null, true);
+    public static void SYSCS_GET_PROGRESS(String drdaToken, Integer getDRDA, final ResultSet[] resultSet) throws SQLException{
```

Comment:
If I understand correctly, SYSCS_GET_PROGRESS returns a formatted progress string. Is it possible to create another function or the same with a parameter to return a int value - progress in percents? It would easier to integrate with other systems, like cloud manager

Reply:
the problem is this
we need the following information:
```
jobNumber numCompletedStages numStages numCompletedTasks numActiveTasks
jobname (may include \n)
```
you can only calculate the percentage complete of **one stage**, by `(numCompletedTasks*100)/numTasks`.
You don't know how many  jobs there are, and often jobs only have one stage.

so, what spark does, it that it will display the following:
`[ Stage $numCompletedStages =======>--------------- ( $numCompletedTasks + $numActiveTasks ) / $numTasks`

e.g.
```
[Stage 2:============================================>              (3 + 1) / 4]
``` 
Now if you have multiple jobs/stages, this is just overwritten in the sparkshell, so there's no global indication of how long this command will take, just an approximation of how long the current stage is.

What GET_PROGRESS will currently do is to return back to the caller all the information we have (jobNumber numCompletedStages numStages numCompletedTasks numActiveTasks jobname), so from there you can generate a progress bar how you like it.
In a UI, where you can do overwrite, i would suggest printing all information like
`[ Job $jobNumber ($jobName), Stage $numCompletedStages / $numStages =======>--------------- ( $numCompletedTasks + $numActiveTasks ) / $numTasks`

You can't overwrite in sqlshell, so we print the `Job $jobNumber ($jobName), Stage $numCompletedStages / $numStages` only if they change, and then we would start a new progress bar.
In the example above, your first job is one where you get a nice progress bar (has a stage with a lot of tasks).
After the first job, you get 10 really small jobs, with each one stage and 2 Tasks. So this is why i added the "smallTasks" boolean. In this case, we don't print a new progress bar for each of these tasks, but combine them together into one progress bar (anyhow, a progressbar with only 2 tasks wouldn't have much information).

The result is 2 nicely progressing progressbars, instead of ~12 where 11 of them aren't really progressing but just finished immediately.

Note that if you're in a UI, you don't have to worry as much about this "smallTasks" problem, as when you can overwrite the UI, you don't have to worry about cluttering the console.

Both approaches require very different information, so this is why i chose to just hand over the raw information.

Reply:
Thank you for the detailed explanation!